### ISSUE TITLE
track multiple wallets

### ISSUE THREAD

**Issue by fsboehme** [Tudmotu/crowsnest#28]
originally titled: "relay investment on transfers between wallets"

when buying with one wallet, then transferring to another for storage, the investment should still be counted. 

probably just need the ability to track multiple wallets for this. otherwise it's hard to know what was a gift/external transfer vs. purchase/internal transfer.

**Comment by Tudmotu** [comment]
What do you think about having a checkbox that says something like "Exclude transfers" and that button hides any NFTs that were not bought directly by the wallet?

**Comment by Tudmotu** [comment]
Actually now I'm reading what you wrote again and I'm not sure I understood correctly.
You mean to let you show a "Combined portfolio" for 2 wallets?
If so - I think it's a cool idea but very complicated. :thinking:

**Comment by fsboehme** [comment]
yeah, i think it would be preferable to allow tracking combined portfolios like that. not sure the checkbox would be enough. how would we distinguish an nft i was sent as a gift or prize (e.g. i won a quiz once and was sent a free nft as the prize) from one I bought with a different wallet (e.g. i bought some with my coinbase wallet and some on niftygateway and then sent them to my metamask)

if i could add my coinbase and niftygateway wallets, then those transfers could be classified as internal and the purchases counted as investments and the free one could be classified as a free acquisition / gift / airdrop / whatever you want to call it.

of course, it would require a little more refactoring... i think, if you want to go down that route, this might also be a good time to add a database backend instead of making all these api calls from scratch every time.

**Comment by Tudmotu** [comment]
Yes, I udnerstand. I like the idea but would require some thought regarding the architecture (which currently assumes a single wallet). I think I'd prefer to leave this for now while focusing on the lower-hanging fruits for starters. There are many trivial features I'd like to get (including sorting, for example).

And about using a backend - I would love to have something more robust, but personally I don't think I will have the time to maintain a backend for this (code, servers, etc). I am not opposed to using a backend different than OpenSea :slightly_smiling_face:  So if you or anyone else would deploy a free NFT API that we can use here, I will definitely consider it!
